pkg/serializer/comments.go

```diff
@@ -117,3 +118,53 @@ func noAnnotationWrapper(metaobj metav1.Object, fn func() error) error {
 	// If the annotation isn't present, just run the function
 	return fn()
 }
+
+// GetCommentSource retrieves the YAML tree used as the source for transferring comments for the given runtime.Object.
+// This may be used externally to implement e.g. re-parenting of the comment source tree when moving structs around.
+func GetCommentSource(obj runtime.Object) (*yaml.RNode, error) {
```

Comment:
Please create some simple unit tests for these functions, and use them in the rest of the code internally